packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -687,12 +706,34 @@ export class DatabaseClusterFromSnapshot extends DatabaseClusterNew {
   constructor(scope: Construct, id: string, props: DatabaseClusterFromSnapshotProps) {
     super(scope, id, props);
 
-    const credentials = renderCredentials(this, props.engine, props.credentials);
-    const secret = credentials.secret;
+    if (props.credentials && !props.credentials.password && !props.credentials.secret) {
+      Annotations.of(this).addWarning('Use `snapshotCredentials` to modify password of a cluster created from a snapshot.');
+    }
+    if (!props.credentials && !props.snapshotCredentials) {
+      Annotations.of(this).addWarning('Generated credentials will not be applied to cluster. Use `snapshotCredentials` instead. `addRotationSingleUser()` and `addRotationMultiUser()` cannot be used on tbis cluster.');
+    }
+    const deprecatedCredentials = renderCredentials(this, props.engine, props.credentials);
+
+    let credentials = props.snapshotCredentials;
+    let secret = credentials?.secret;
+    if (!secret && credentials?.generatePassword) {
+      if (!credentials.username) {
+        throw new Error('`snapshotCredentials` `username` must be specified when `generatePassword` is set to true');
+      }
+
+      secret = new DatabaseSecret(this, 'SnapshotSecret', {
+        username: credentials.username,
+        encryptionKey: credentials.encryptionKey,
+        excludeCharacters: credentials.excludeCharacters,
+        replaceOnPasswordCriteriaChanges: credentials.replaceOnPasswordCriteriaChanges,
+        replicaRegions: credentials.replicaRegions,
+      });
+    }
 
     const cluster = new CfnDBCluster(this, 'Resource', {
       ...this.newCfnProps,
       snapshotIdentifier: props.snapshotIdentifier,
+      masterUserPassword: secret?.secretValueFromJson('password')?.unsafeUnwrap() ?? credentials?.password?.unsafeUnwrap(), // Safe usage
```

Comment:
This is the functionality we're trying to avoid, since [CloudFormation doesn't support](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-masteruserpassword) setting this property when also setting the `snapshotIdentifier`

It deploys - but from my testing last week, the `masterUserPassword` prop ends up getting ignored and as a result the generated secret has an incorrect password. I had to manually change the password of the secret which was generated to be the secret which the initial db was created with to be able to login. 

Were you able to successfully login to the DB after creating a cluster this way? I'm also curious if you were able to login to the DB after rotating credentials.

Reply:
The documentation is incorrect. It is supported, it works exactly like for a DB instance (where the doc also says that the master password should not be specified for a snapshot). For a cluster you need to wait until all instances finish deploying before RDS changes the password.

Rotation is the confirmation that it works. It only  succeeds (= sets the new password in Secrets Manager) if it can log in to the cluster (see [code](https://github.com/aws-samples/aws-secrets-manager-rotation-lambdas/blob/master/SecretsManagerRDSMySQLRotationSingleUser/lambda_function.py)])

You can deploy my integ test to test this.